run/fvde2john.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python3
+
+# Usage: python3 fvde2john.py <image_file>
+# The partition table is parsed to find the boot volume, often named 'Recovery HD'. The boot volume can be identified by its type GUID: 426F6F74-0000-11AA-AA11-00306543ECAC.
+# The boot volume contains a file called `EncryptedRoot.plist.wipekey`. This is stored on the volume at `/com.apple.boot.X/System/Library/Caches/com.apple.corestorage/EncryptedRoot.plist.wipekey`, where `X` is variable but is often `P` or `R`. This plist file is encrypted with AES-XTS; the key is found in the CoreStorage volume header, and the tweak is b'\x00' * 16.
+# The decrypted plist contains information relating to the user(s). This includes the salt, kek and iterations required to construct the hash as well as information such as username and password hints (if present).
+
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+import plistlib
+import os
+import argparse
+import pytsk3    # pip install pytsk3
+
+HEX_CORE_STORAGE_TYPE_GUID = '53746F72-6167-11AA-AA11-00306543ECAC'
+HEX_APPLE_BOOT_STORAGE_TYPE_GUID = '426F6F74-0000-11AA-AA11-00306543ECAC'
+LOCAL_USER_TYPE_ID = 0x10060002
+
+def uint_to_int(b):
+    return int(b[::-1].hex(), 16)
+
+def guid_to_hex(guid):
+    guid_parts = guid.split('-')
+
+    hex_str = ''.join([guid_parts[0][i:i+2] for i in range(0, len(guid_parts[0]), 2)][::-1])
+    hex_str += ''.join([guid_parts[1][i:i+2] for i in range(0, len(guid_parts[1]), 2)][::-1])
+    hex_str += ''.join([guid_parts[2][i:i+2] for i in range(0, len(guid_parts[2]), 2)][::-1])
+    hex_str += guid_parts[3]
+    hex_str += guid_parts[4]
+
+    return hex_str.lower()
+
+def parse_partition_table(fp):
+    # determine whether sector size is 0x200 or 0x1000
+    sector_size = 0x0
+
+    # look for EFI PART at start of sector 1
+    fp.seek(0x200)
+    signature = fp.read(0x8)
+    if signature == b'EFI PART':
+        sector_size = 0x200
+
+    else:
+        fp.seek(0x1000)
+        signature = fp.read(0x8)
+        if signature == b'EFI PART':
+            sector_size = 0x1000
+
+    print("[+] Identified sector size:", sector_size)
+
+    if not sector_size:
+        print(f"[!] Invalid sector size {sector_size} (not 512 or 4096 bytes). Exiting.")
```

Comment:
It says it will exit but it doesn't. A `sys.exit(1)` is missing.

```suggestion
        print(f"[!] Invalid sector size {sector_size} (not 512 or 4096 bytes). Exiting.")
        sys.exit(1)
```